src/Form/Type/ArticleSelectionElementType.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz's Blog plugin for Sylius.
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusBlogPlugin\Form\Type;
+
+use MonsieurBiz\SyliusBlogPlugin\Entity\Article;
+use MonsieurBiz\SyliusBlogPlugin\Entity\ArticleInterface;
+use MonsieurBiz\SyliusBlogPlugin\Repository\ArticleRepositoryInterface;
+use Sylius\Bundle\ResourceBundle\Form\DataTransformer\ResourceToIdentifierTransformer;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\ReversedTransformer;
+use Symfony\Component\Validator\Constraints as Assert;
+
+final class ArticleSelectionElementType extends AbstractType
+{
+    public function __construct(
+        private readonly ArticleRepositoryInterface $articleRepository,
+        private readonly ChannelContextInterface $channelContext,
+        private readonly LocaleContextInterface $localeContext,
+    ) {
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $articles = $this->articleRepository->createShopListQueryBuilderByType(
+            $this->localeContext->getLocaleCode(),
+            ArticleInterface::BLOG_TYPE,
+            $this->channelContext->getChannel(),
+            null
+        );
+
+        $articles = $articles->orderBy('translation.title')->getQuery()->getResult();
+
+        $builder
+            ->add('article', EntityType::class, [
+                'class' => Article::class,
+                'label' => 'monsieurbiz_blog.ui_element.articles_selection_ui_element.fields.article',
+                'choice_label' => fn (Article $article) => $article->getTitle(),
+                'choice_value' => fn (?Article $article) => $article?->getId(),
+                'required' => true,
+                'choices' => $articles,
+            ])
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:**  I don't think this is the best way to give the list of choices, you should use `query_builder` as in https://github.com/monsieurbiz/SyliusBlogPlugin/blob/master/src/Form/Type/ArticleType.php#L69‏

Reply:
@delyriand I took it from https://github.com/monsieurbiz/SyliusBlogPlugin/blob/master/src/Form/Type/CaseStudyElementType.php 😢

Reply:
in this case, can you replace this formtype also 😘

Reply:
@maximehuran Are you ok if I make these suggested changes on the existing class ?

Reply:
If it's cleaner it's better so let's go 🚀

Reply:
I changed it @delyriand @maximehuran